test/optimizer/selectivity_util_test.cpp

```diff
@@ -0,0 +1,473 @@
+    for (int i = 0; i < vals_3.size(); ++i) {
+      for (int j = 0; j < vals_3[i].second; ++j) {
+        column_stats_obj_3_.GetHistogram()->Increment(vals_3[i].first);
+      }
+    }
+
+    // DECIMAL column.
+    column_stats_obj_4_ =
+        NewColumnStats<execution::sql::Real>(catalog::db_oid_t(1), catalog::table_oid_t(1), catalog::col_oid_t(4),
+                                             600000, 500500, 0.1658, 10, 500, 10, true);
+    // Assume entry with value i occurs i times in the table.
+    for (int i = 1; i <= 1000; ++i) column_stats_obj_4_.GetTopK()->Increment(static_cast<float>(i), i);
+    // Construct histogram.
+    for (int i = 0; i <= 1000; ++i) {
+      for (int j = 0; j < i; ++j) {
+        column_stats_obj_4_.GetHistogram()->Increment(i);
+      }
+    }
+
+    // BOOLEAN column.
+    column_stats_obj_5_ = NewColumnStats<execution::sql::BoolVal>(
+        catalog::db_oid_t(5), catalog::table_oid_t(5), catalog::col_oid_t(5), 100, 80, 0.2, 10, 10, 10, true);
+    // 60 true, 20 false, 20 null.
+    column_stats_obj_5_.GetTopK()->Increment(true, 60);
+    column_stats_obj_5_.GetTopK()->Increment(false, 20);
+
+    // TODO(arvindsk) The histogram class currently does not support VARCHAR entries. Templatize the histogram class.
+//    column_stats_obj_6_ = NewColumnStats<execution::sql::StringVal>(
+//        catalog::db_oid_t(6), catalog::table_oid_t(6), catalog::col_oid_t (6), 10, 5, 0.5, 10, 10, 10, true);
+//    // Values and frequencies.
+//    std::vector<std::pair<std::string_view, int>> vals_6 = {{"hello", 2}, {"abc", 1}, {"def", 1}, {"xyz", 1}};
+//
+//    // Construct Top k variable.
+//    for (int i = 0; i < vals_6.size(); ++i) {
+//      storage::VarlenEntry value;
+//      value.Create(vals_6[i].first);
+//      column_stats_obj_6_.GetTopK()->Increment(value, vals_6[i].second);
+//    }
+//
+//    // Construct histogram.
+//    for (int i = 0; i < vals_3.size(); ++i) {
+//      for (int j = 0; j < vals_3[i].second; ++j) {
+//        column_stats_obj_3_.GetHistogram()->Increment(vals_3[i].first);
+//      }
+//    }
+  }
+};
+
+// NOLINTNEXTLINE
+TEST_F(SelectivityUtilTests, TestFloatLessThan) {
+  // TEST PART 1
+  // Create a constant value expression to pass to ValueCondition.
+  // Floating point numbers have DECIMAL type and are represented as Reals in execution layer.
+  std::unique_ptr<parser::ConstantValueExpression> const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::DECIMAL, execution::sql::Real(6.f));
+
+  // Create a value condition to pass to SelectivityUtil.
+  ValueCondition value_condition(catalog::col_oid_t(1), parser::ExpressionType::COMPARE_LESS_THAN,
+                                 std::move(const_value_expr_ptr));
+
+  // Compute selectivity for col < 6.
+  double res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Real>>(&column_stats_obj_1_), value_condition);
+
+  ASSERT_DOUBLE_EQ(res, 1.f);
```

Comment:
Recheck this assert statement, I don't think you consider the fraction null and the answer should be 0.8